src/NetTopologySuite/Geometries/Position.cs

```diff
@@ -0,0 +1,55 @@
+namespace NetTopologySuite.Geometries
+{
+    /// <summary>
+    /// Indicates the position of a location relative to a
+    /// node or edge component of a planar topological structure.
+    /// </summary>
+    public enum Position
```

Comment:
What I like about enums is that I don't just see the value but also its meaning.   
Maybe we can mix both, rename `Position` enum to `Position[Index|Value]` and then use that inside Position enum-class.
What is the benefit of an enum-class over a struct in this case?

Reply:
>  What is the benefit of an enum-class over a struct in this case?

It could be a struct instead, the point I wanted to make is just that a .NET `enum` can't have methods or properties, so in trying to port one Java class this way, we wind up making two .NET types, with one of them feeling out of place.

In most places, I also favor designing around .NET `enum`s, but obviously our design is more constrained in core NTS.

Reply:
I refactored `Position` to a struct.